src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java

```diff
@@ -228,103 +228,12 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
     }
 
-    /**
-     * Does the Job have _all_ the needed items.
-     *
-     * @return true if the Job has no needed items.
-     */
-    public boolean isMissingNeededItem()
+    public Set<IToken> getAsyncRequests()
     {
-        return !itemsNeeded.isEmpty();
-    }
-
-    /**
-     * Get the list of items needed by the Job.
-     *
-     * @return List of items needed by the Job.
-     */
-    @NotNull
-    public List<ItemStack> getItemsNeeded()
-    {
-        return Collections.unmodifiableList(itemsNeeded);
-    }
-
-    /**
-     * Reset the items needed.
-     */
-    public void clearItemsNeeded()
-    {
-        itemsNeeded.clear();
-    }
-
-    /**
-     * Add (or increment) an ItemStack to the items needed by the Job.
-     * We're not comparing item damage values since i.e a damaged rod is the same as a normal rod for our purpose.
-     *
-     * @param stack Item+count needed to do the job.
-     */
-    public void addItemNeeded(@NotNull final ItemStack stack)
-    {
-        for (@NotNull final ItemStack neededItem : itemsNeeded)
-        {
-            if (stack.isItemEqualIgnoreDurability(neededItem))
-            {
-                ItemStackUtils.changeSize(neededItem, ItemStackUtils.getSize(stack));
-                return;
-            }
-        }
-
-        itemsNeeded.add(stack.copy());
-    }
-
-    /**
-     * Remove a items from those required to do the Job.
-     * We're not comparing item damage values since i.e a damaged rod is the same as a normal rod for our purpose.
-     *
-     * @param stack ItemStack (item+count) to remove from the list of needed items.
-     * @return modified ItemStack with remaining items (or null).
-     */
-    @Nullable
-    public ItemStack removeItemNeeded(@NotNull final ItemStack stack)
-    {
-        @NotNull final ItemStack stackCopy = stack.copy();
-
-        if (stack.isEmpty())
-        {
-            return stackCopy;
-        }
-
-        for (@NotNull final ItemStack neededItem : itemsNeeded)
-        {
-            if (stack.isItemEqualIgnoreDurability(neededItem))
-            {
-                //todo make this sofisticated as soon as material handling has been implemented.
-                //final int itemsToRemove = Math.min(ItemStackUtils.getSize(neededItem), ItemStackUtils.getSize(stackCopy));
-                //ItemStackUtils.changeSize(neededItem, -itemsToRemove);
-                //ItemStackUtils.changeSize(stackCopy, -itemsToRemove);
-
-                //Deativate this if for now in order to keep working even if not all items are given. previously checked if stackSize is 0 and only removed then.
-                itemsNeeded.remove(neededItem);
-
-                break;
-            }
-        }
-
-        return ItemStackUtils.isEmpty(stackCopy) ? ItemStackUtils.EMPTY : stackCopy;
+        return asyncRequests;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "asyncRequests". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2384)